sklearn/datasets/tests/test_samples_generator.py

```diff
@@ -61,6 +61,18 @@ def test_make_classification():
                  .reshape(-1, X.shape[1]).shape[0], 2000,
                  "Unexpected number of unique rows")
 
+    # Test to ensure that weights are not modified
```

Comment:
A personal suggestion for your reference. Maybe we don't need a separate test for such a small issue. We might integrate current test to previous test. e.g.
```python
weights = [0.1, 0.25]
X, y = make_classification(n_samples=100, n_features=20, n_informative=5,
                           n_redundant=1, n_repeated=1, n_classes=3,
                           n_clusters_per_class=1, hypercube=False,
                           shift=None, scale=None, weights=weights,
                           random_state=0)
assert_equal(weights, [0.1, 0.25])
assert_equal(X.shape, (100, 20), "X shape mismatch")
assert_equal(y.shape, (100,), "y shape mismatch")
assert_equal(np.unique(y).shape, (3,), "Unexpected number of classes")
assert_equal(sum(y == 0), 10, "Unexpected number of samples in class #0")
assert_equal(sum(y == 1), 25, "Unexpected number of samples in class #1")
assert_equal(sum(y == 2), 65, "Unexpected number of samples in class #2")

Reply:
Yes, done!